src/gamepad.cpp

```diff
@@ -81,7 +81,6 @@ void Gamepad::setup()
 		}
 	}
 
-	lastAction = HOTKEY_NONE;
```

Comment:
Why this removal? It is necessary to keep hotkeys that would cycle or toggle through something if held (such as this one, incidentally) from doing something unpredictable. I don't think this should be removed.

Reply:
That line would apply HOTKEY_NONE to lastAction immediately after the profile was loaded, causing it to rapidly apply the button's function dozens of times every second while the hotkey button is held down. This appeared to also happen with other hotkeys that used it, such as switching to a static profile, although we wouldn't notice it since applying profile 4 ten times still gives you profile 4. After I removed it it allowed me to press the button once to get it to apply only once. 

Although I think you are still correct, while it did seem to be working well for me I'm going to revisit the code to see if there is some other function that is setting it to HOTKEY_NONE and causing it to work correctly accidentally.

Reply:
If you find the first paragraph to be true when it's removed, then I think a refactor changed the behavior somehow. You may be right, so try more and I can try to take a look at main some time today too.

Either way though, indeed the behavior of this hotkey and other hotkeys like toggle 4-way should be that you hold it for any length of time and trigger it once, so focus on that and if you confirm the line should go, then that's a good find.

Reply:
I looked at the code again and it appears that this module is called every loop of the main firmware loop. It overwrites the lastAction variable as long as there isn't a hotkey being pressed. So I feel that the line I removed is redundant. https://github.com/OpenStickCommunity/GP2040-CE/blob/affc3c498a37b558967bff91df2209103606df30/src/gamepad.cpp#L239-L266

The only issue I've noticed is that there is about a 1/4th of a second of delay before I can press the profile button again. But I don't see another spot that I could edit it to make it faster.

Reply:
250 ms doesn't make much sense to me, but maybe the reinitialization blocks things for a bit; regardless I agree with your assessment otherwise.